File: longest_common_prefix/longest_common_prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return "" 
        lcp = strs[0]
        for str in strs: 
            i=0
            while i < lcp.length: 
                if(str[i] != lcp[i]):
                    lcp = lcp[0:i]
        return lcp



    print(longestCommonPrefix(self,["asdf","awer","axcv"]))

File: longest_common_prefix.rb
# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
    return "" if strs.empty?
    lcp = strs[0] 
    strs.each do |str| 
        lcp.each_char.with_index do |char, i| 
             if(str[i] != lcp[i])
                 lcp = lcp[0...i]
             end
        end
    end
    
    return lcp
end 

print(longest_common_prefix(["flower","f", "flow", "flock"]))
print("\n")